### PR TITLE
elantp: Fix an unset i2c device name

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2238139

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
FWIW, this was the device:

	07:53:25.158 FuEngine             /sys/devices/pci0000:00/0000:00:15.1/i2c_designware.1/i2c-1/i2c-VEN_04F3:00/0018:04F3:311C.0001/hidraw/hidraw1 added FuUdevDevice:
	  Name:                 VEN 04F3:00 04F3:311C
	  Guid:                 46598f8d-5777-50c4-8da9-cda0cc8bbc26 <- HIDRAW\VEN_04F3&DEV_311C ?
	  Flags:                internal
	  VendorId:             HIDRAW:0x04F3
	  Guid[quirks]:         d8719940-e927-5fea-8fad-a176f01d9ff2 <- HIDRAW\VEN_04F3
	  BackendId:            /sys/devices/pci0000:00/0000:00:15.1/i2c_designware.1/i2c-1/i2c-VEN_04F3:00/0018:04F3:311C.0001/hidraw/hidraw1
	  AcquiesceDelay:       2500
	  PossiblePlugin:       elantp
	  Vendor:               0x4f3
	  Model:                0x311c
	  Subsystem:            hidraw
	  DeviceFile:           /dev/hidraw1
	  SysfsPath:            /sys/devices/pci0000:00/0000:00:15.1/i2c_designware.1/i2c-1/i2c-VEN_04F3:00/0018:04F3:311C.0001/hidraw/hidraw1

File: plugins/elantp/fu-elantp-hid-device.c [hunk 1]
```diff
@@ -433,6 +433,11 @@ fu_elantp_hid_device_setup(FuDevice *device, GError **error)
 		g_autoptr(FuElantpHidHapticDevice) cfg = fu_elantp_haptic_device_new(device);
 		fu_device_add_child(FU_DEVICE(device), FU_DEVICE(cfg));
 	}
+
+	/* fix an unsuitable i²c name, e.g. `VEN 04F3:00 04F3:311C` */
+	if (g_str_has_prefix(fu_device_get_name(device), "VEN "))
+		fu_device_set_name(device, "Touchpad");
```
Comment (superm1) [inline plugins/elantp/fu-elantp-hid-device.c:439]: How do you know it's not a touchscreen?  I think it's better to just call it "Human Interface Device" if you don't know.

Comment (hughsie) [reply plugins/elantp/fu-elantp-hid-device.c:439]: @jinglewu can you help here? Thanks!

Comment (jinglewu) [reply plugins/elantp/fu-elantp-hid-device.c:439]: Elan touchpad VID 04f3 PID 3XXX.

Comment (hughsie) [reply plugins/elantp/fu-elantp-hid-device.c:439]: Thanks, patch updated.

**Review by superm1 — APPROVED** [review]

**Pull request merged**